rcl/src/rcl/wait.c

```diff
@@ -83,7 +83,10 @@ __wait_set_is_valid(const rcl_wait_set_t * wait_set)
 static void
 __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
 {
-  if (wait_set->subscriptions) {
+  if (
+    wait_set->subscriptions || wait_set->guard_conditions ||
+    wait_set->timers || wait_set->clients || wait_set->services)
```

Comment:
It looks like `rcl_wait_set_resize` does these checks internally, so the `if` statement could be removed altogether.

https://github.com/ros2/rcl/blob/26744b0e5d71790b9b4bd376138f5b1a6ae15e66/rcl/src/rcl/wait.c#L264

Reply:
Sounds even better: 20fdf1b5b82a6b1dc23cc7bef643bf18dfdcd2c3